Metalama.Framework.Engine/Linking/LinkerInjectionStep.Rewriter.cs

```diff
@@ -283,26 +289,41 @@ private T AddSuppression<T>( T node, IReadOnlyList<string> suppressionsOnThisEle
                     // If we are removing a custom attribute, keep its trivia.
                     foreach ( var trivia in list.GetLeadingTrivia() )
                     {
-                        switch ( trivia.Kind() )
+                        if ( trivia.Kind() is SyntaxKind.SingleLineCommentTrivia 
+                                or SyntaxKind.MultiLineCommentTrivia 
+                                or SyntaxKind.SingleLineDocumentationCommentTrivia 
+                                or SyntaxKind.MultiLineDocumentationCommentTrivia )
                         {
-                            case SyntaxKind.MultiLineCommentTrivia or SyntaxKind.MultiLineDocumentationCommentTrivia:
-                                outputTrivia.Add( trivia );
-
-                                break;
+                            // Doc comments from the 
+                            if ( wasFirstList )
+                            {
+                                firstListLeadingTrivia ??= new List<SyntaxTrivia>();
+                                firstListLeadingTrivia.Add( trivia );
 
-                            case SyntaxKind.SingleLineCommentTrivia or SyntaxKind.SingleLineDocumentationCommentTrivia:
+                                if ( trivia.Kind() is SyntaxKind.SingleLineCommentTrivia or SyntaxKind.SingleLineDocumentationCommentTrivia )
+                                {
+                                    firstListLeadingTrivia.Add( ElasticLineFeed );
+                                }
+                            }
+                            else
+                            {
                                 outputTrivia.Add( trivia );
-                                outputTrivia.Add( ElasticLineFeed );
```

Comment:
I think this is still necessary for `SingleLine` trivias, even when it's not the first list.